### PR TITLE
openzfs restructuring part 1

### PR DESCRIPTION
**Pull request by mattmacy**
openzfs restructuring part 1
    move platform specific headers

    Create both a user and kernel space sys/simd.h header which can
    be included in either context.  No functional change, the source
    has been refactored and the relevant #include's updated.
    
    
    Signed-off-by: Matthew Macy <mmacy@FreeBSD.org>
    Signed-off-by: Brian Behlendorf <behlendorf1@llnl.gov>

This is the first part of an effort to break out all the non-functional changes from 8987

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [x] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [x] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [x] All new and existing tests passed.
- [x] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: include/Makefile.am [hunk 1]
```diff
@@ -1,4 +1,10 @@
-SUBDIRS = linux spl sys
+SUBDIRS = sys
+SUBDIRS+= os
```
Comment (ikozhukhov) [inline include/Makefile.am]: i'd like to add space after SUBDIRS:
SUBDIRS += os

**Comment by ikozhukhov** [comment]
i'd like do not move primary ZFS headers - like include/sys/zio.h, etc - to platform specific directories - they are should be the same for all platforms.
spl is more platform specific, but we need a structure with shared headers + platform specific.

**Comment by behlendorf** [comment]
@lundman @ikozhukhov  would you mind taking another look at this.  The updated version additionally:

* Moves the existing `include/linux` directory to `include/os/linux/kernel/linux`.  It should only contain linux-kernel compatibility code, and is not needed by other platforms.
* Refactors the SIMD code slightly by moving the public interfaces out of the `linux` subdirectory and in to the `spl` or `libspl` code.

There might still need to be a few additional tweaks since I haven't yet had a chance to verify this change doesn't break building Lustre against ZFS.

**Comment by lundman** [comment]
We got `simd.h` from ZOL and it has changed a fair bit since then, but it looks like it should work.

**Comment by mattmacy** [comment]
> We got `simd.h` from ZOL and it has changed a fair bit since then, but it looks like it should work.

`simd.h` is part of the SPL - FreeBSD’s is quite different

**Pull request merged**